src/models/resolver.ts

```diff
@@ -9,12 +9,30 @@ import HandlerInstance from "@/models/handler.instance.js";
  * 
  * @export
  * @class Resolver
- * @implements {IResolver}
+ * @implements {Resolver}
  */
-export default class Resolver implements IResolver {
-    // Contains the mapping of the functions
+export default class Resolver implements ResolverInterface {
+    private static _instance: ResolverInterface | undefined;
+
     private _instances: HandlerInstance[] = [];
 
+    public static get instance() {
+        if(this._instance) {
+            return this._instance;
+        }
+
+        const instance = new Resolver();
+        this._instance = instance;
+
+        return instance;
+    }
+
+    public static set instance(value: ResolverInterface) {

```

Comment:
I think that if we choose _Singleton_ we must adhere to it and do not permit to change

Reply:
Agreed. Then I would use Singleton for most of it. Because the problem is that that's technically implicitly already what's happening inside the "requestHandler" decorations added on the classes. It expects one instance.